commit 8c8cddbe022af5ece9949804cacffbee6b8972fa
Author: nicm <nicm>
Date:   Tue Oct 20 14:19:27 2015 +0000

    The table could change when retrying so don't save it at start of
    server_client_handle_key.

--- a/server-client.c
+++ b/server-client.c
@@ -540,7 +540,7 @@
 	struct window		*w;
 	struct window_pane	*wp;
 	struct timeval		 tv;
-	struct key_table	*table = c->keytable;
+	struct key_table	*table;
 	struct key_binding	 bd_find, *bd;
 	int			 xtimeout;
 
@@ -606,7 +606,7 @@
 retry:
 	/* Try to see if there is a key binding in the current table. */
 	bd_find.key = key;
-	bd = RB_FIND(key_bindings, &table->key_bindings, &bd_find);
+	bd = RB_FIND(key_bindings, &c->keytable->key_bindings, &bd_find);
 	if (bd != NULL) {
 		/*
 		 * Key was matched in this table. If currently repeating but a
@@ -624,6 +624,7 @@
 		 * Take a reference to this table to make sure the key binding
 		 * doesn't disappear.
 		 */
+		table = c->keytable;
 		table->references++;
 
 		/*
